src/server/cluster/cluster_family.cc

```diff
@@ -504,11 +504,12 @@ void ClusterFamily::DflyClusterConfig(CmdArgList args, ConnectionContext* cntx)
 
   auto cb = [&](util::ProactorBase* pb) { tl_cluster_config = new_config; };
   server_family_->service().proactor_pool().AwaitFiberOnAll(std::move(cb));
-
   DCHECK(tl_cluster_config != nullptr);
 
-  SlotSet after = tl_cluster_config->GetOwnedSlots();
+  bool success = server_family_->AwaitCurrentDispatches(absl::Seconds(1), cntx->conn());
+  // TODO... and what to do?
```

Comment:
@chakaz, @adiholden 

Should there be a timeout during a cluster config push? (i.e. waiting for a command that is potentially still using the old config)

If so, this is really only likely with blocking commands (or someone runs scripts that take 1s), should we cancel blocking commands?

Reply:
That's a very good question.
Blocking commands specifically can never end, so I guess we should be cancelling them. Not sure about 1s though, it might be a bit too short?
@adiholden wdyt?

Reply:
we should cancel blocking commands with an INTERRUPTED error or similar.

Reply:
Another issue is that we also might be not able to revert the changes, if another endless command starts running on a new slot.

For example, If I start `EVAL while true; do ... ; end` and we fail to switch the config, we will also fail to revert it back (even with a sophisticated solution that tracks slots, because I can start the same script on different slots now). It's an in practice impossible scenario, but still

Reply:
We could just be doing a client pause actually at the start, thought with most cases with simple commands its not necessary to interrupt traffic at all to do a config switch